CMakeLists.txt

```diff
@@ -145,6 +145,12 @@ if(BUILD_TESTING)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
   endif()
 
+  # BENCHMARK_F seems to break cppcheck 2.1
+  # set(ament_cmake_cppcheck_FOUND TRUE)
```

Comment:
Is this just disabling cppcheck? I don't think that's an ok solution. If it is only disabling it for the tests somehow that would be ok I think. You can also disable cppcheck locally, e.g.:

https://github.com/ros2/rclcpp/blob/c214ddc57b1f2e75076aa087e77c9a06b769008d/rclcpp/include/rclcpp/node.hpp#L195

Reply:
This was to disable cppcheck, I copied it from the performance fixture test demo. Although I enabled it locally and it seems to be working fine. I'll run CI and check if `cppcheck` causes problems with this PR.